otomoto/spiders/otomoto.py

```diff
@@ -2,34 +2,44 @@
 
 from scrapy.loader import ItemLoader
 from scrapy.loader.processors import TakeFirst, MapCompose, Compose
-
+from urllib.parse import urlparse
 from otomoto.items import OtomotoItem
 
-
 def filter_out_array(x):
     x = x.strip()
     return None if x == '' else x
 
-
 def remove_spaces(x):
     return x.replace(' ', '')
 
-
 def convert_to_integer(x):
-    return int(x)
-
+    price = x.replace(',', '.')
+    return int(price)
+
+def convert_vin(x):
+    if "?vin=" in x:
+        vin = (urlparse(x).query.split('vin=', 1)[1]).split('&', 1)[0]
+        if len(vin) != 17:
+            vin = 'null'
+    else:
+        vin = 'null'
+    return vin
```

Comment:
This still could be simplified.  Check out docs how you can extract query parameter from URL easy way